packages/expo-haptics/ios/HapticsModule.swift

```diff
@@ -4,26 +4,16 @@ public class HapticsModule: Module {
   public func definition() -> ModuleDefinition {
     name("ExpoHaptics")
 
-    method("notificationAsync") { (notificationType: String, promise: Promise) in
-      guard let feedbackType = NotificationType(rawValue: notificationType)?.toFeedbackType() else {
-        promise.reject("E_HAPTICS_INVALID_ARG", "Notification type must be one of: 'success', 'warning', 'error'. Obtained '\(notificationType)'")
-        return
-      }
+    method("notificationAsync") { (notificationType: NotificationType) in
```

Comment:
What has happened with `promise: Promise` parameter? 🤔

Reply:
There is no need for it to be there — the function is not async (it's only exported as async). The `ConcreteMethod` detects whether the `Promise` is the last argument and if not, it automatically resolves once the method returns.
Previously it's been there because `promise.reject` was used 😉

Reply:
Another wild idea: what would you say about hiding the `promise` altogether in every method (optionally you can define it as the last parameter)?
Let's say you simply return a value or throw an exception in the method body and SweetAPI decides whether it's async method with promise or not 🤔

Reply:
Isn't it exactly what I just said? 😉 That's how it works now.

Reply:
Awesome 👏 I can't wait to use it 😊 
I assumed this auto-promise-wrapping mechanism works only for `void` return type 😉